### PR TITLE
Fetch manifest if not present during `system stop` for kubernetes

### LINKED ISSUE #3897: [mesheryctl] `system stop` fails for Kubernetes platform if the manifest files are not present
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior

`system stop` throws an error that the manifest file is missing if the `.meshery/manifests` folder is missing the manifests and stops executing.

#### Desired Behavior

If the files are missing, then fetch the corresponding version of the manifests as specified in the meshconfig(`.meshery/config.yaml`) file and stop Meshery.

---

#### Contributor Resources
- [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by suhaniii**
Signed-off-by: Suhani Agarwal <suhani2612@gmail.com>

**Description**

This PR fixes #3897 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/system/stop.go [hunk 1]
```diff
@@ -141,6 +141,19 @@ func stop() error {
 			return err
 		}
 
+		// check if deployments exists in manifest folder
+		if _, err := os.Stat(filepath.Join(utils.MesheryFolder, utils.ManifestsFolder, utils.MesheryDeployment)); os.IsNotExist(err) {
+			_, err = utils.FetchManifests(currCtx)
+			if err != nil {
+				return errors.Wrap(err, "Unable to fetch deployment")
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/stop.go:148]:
```suggestion
				return errors.Wrap(err, "Unable to fetch Meshery deployment manifests")
```

File: mesheryctl/internal/cli/root/system/stop.go [hunk 2]
```diff
@@ -141,6 +141,19 @@ func stop() error {
...
+			}
+			// Download operator manifest
+			err = utils.DownloadOperatorManifest()
+			if err != nil {
+				return ErrDownloadFile(err, "operator manifest")
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/stop.go:153]:
```suggestion
				return ErrDownloadFile(err, "Meshery Operator manifest")
```

File: mesheryctl/internal/cli/root/system/stop.go [hunk 3]
```diff
@@ -141,6 +141,19 @@ func stop() error {
 			return err
 		}
 
+		// check if deployments exists in manifest folder
+		if _, err := os.Stat(filepath.Join(utils.MesheryFolder, utils.ManifestsFolder, utils.MesheryDeployment)); os.IsNotExist(err) {
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/stop.go:145]: @suhaniii is this logic considering the specific version of Meshery that the current-context is using?

Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/stop.go:145]: @piyushsingariya @metonymic-smokey @navendu-pottekkat are we on a path to having a structured set of folder-versioned manifests?

Comment (metonymic-smokey) [reply mesheryctl/internal/cli/root/system/stop.go:145]: @leecalcote we have a design spec for air gapped deployments. We could add considerations for it there, I'll work towards adding them.

Comment (suhaniii) [reply mesheryctl/internal/cli/root/system/stop.go:145]: @leecalcote  It is just checking if `MesheryDeployment` is present during `system stop`. 
What else do I need to consider?

Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/stop.go:145]: Thank you, @metonymic-smokey. Wonderful.

Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/stop.go:145]: @suhaniii, I think I'm jumping the gun with my remarks and that your update here is inline with the current architecture. Good stuff. ✔️

**Review by leecalcote — APPROVED** [review]

**Pull request merged**